src/main/java/bio/terra/cli/businessobject/resources/BqDataset.java

```diff
@@ -109,6 +103,26 @@ public static BqDataset createControlled(CreateUpdateBqDataset createParams) {
     return new BqDataset(createdResource);
   }
 
+  /** Update a Big Query dataset resource in the workspace. */
+  public void update(UpdateResourceParams updateParams) {
+    if (updateParams.name != null) {
+      validateEnvironmentVariableName(updateParams.name);
+    }
+    switch (stewardshipType) {
+      case REFERENCED:
+        WorkspaceManagerService.fromContext()
+            .updateReferencedBigQueryDataset(Context.requireWorkspace().getId(), id, updateParams);
+        break;
+      case CONTROLLED:
+        WorkspaceManagerService.fromContext()
+            .updateControlledBigQueryDataset(Context.requireWorkspace().getId(), id, updateParams);
+        break;
+      default:
+        throw new IllegalArgumentException("Unknown stewardship type: " + stewardshipType);
+    }
+    super.update(updateParams);
```

Comment:
it's a little confusing to me that the public method that makes a workspace manager call & the protected method for modifying the resource & state and caching have the same name. Can we differentiate between updating the workspace manager & updating our cached resources?

Reply:
renamed the superclass method.